jupyter_client/tests/test_kernelmanager.py

```diff
@@ -127,3 +128,122 @@ def test_start_new_kernel(self):
 
         self.assertTrue(km.is_alive())
         self.assertTrue(kc.is_alive())
+
+
+class TestAsyncKernelManager(AsyncTestCase):
+    def setUp(self):
+        super(TestAsyncKernelManager, self).setUp()
+        self.env_patch = test_env()
+        self.env_patch.start()
+
+    def tearDown(self):
+        super(TestAsyncKernelManager, self).tearDown()
+        self.env_patch.stop()
+
+    def _install_test_kernel(self):
+        kernel_dir = pjoin(paths.jupyter_data_dir(), 'kernels', 'signaltest')
+        os.makedirs(kernel_dir)
+        with open(pjoin(kernel_dir, 'kernel.json'), 'w') as f:
+            f.write(json.dumps({
+                'argv': [sys.executable,
+                         '-m', 'jupyter_client.tests.signalkernel',
+                         '-f', '{connection_file}'],
+                'display_name': "Signal Test Kernel",
+            }))
+
+    def _get_tcp_km(self):
+        c = Config()
+        km = AsyncKernelManager(config=c)
+        return km
+
+    def _get_ipc_km(self):
+        c = Config()
+        c.KernelManager.transport = 'ipc'
+        c.KernelManager.ip = 'test'
+        km = AsyncKernelManager(config=c)
+        return km
+
+    @gen.coroutine
+    def _run_lifecycle(self, km):
+        yield km.start_kernel(stdout=PIPE, stderr=PIPE)
+        is_alive = yield km.is_alive()
+        self.assertTrue(is_alive)
+        yield km.restart_kernel(now=True)
+        is_alive = yield km.is_alive()
+        self.assertTrue(is_alive)
+        yield km.interrupt_kernel()
+        self.assertTrue(isinstance(km, AsyncKernelManager))
+        yield km.shutdown_kernel(now=True)
+
+    @gen_test
+    def test_tcp_lifecycle(self):
+        km = self._get_tcp_km()
+        yield self._run_lifecycle(km)
+
+    @skip_win32
+    @gen_test
+    def test_ipc_lifecycle(self):
+        km = self._get_ipc_km()
+        yield self._run_lifecycle(km)
+
+    def test_get_connect_info(self):
+        km = self._get_tcp_km()
+        cinfo = km.get_connection_info()
+        keys = sorted(cinfo.keys())
+        expected = sorted([
+            'ip', 'transport',
+            'hb_port', 'shell_port', 'stdin_port', 'iopub_port', 'control_port',
+            'key', 'signature_scheme',
+        ])
+        self.assertEqual(keys, expected)
+
+    @skip_win32
+    @gen_test
+    def test_signal_kernel_subprocesses(self):
+        self._install_test_kernel()
+        km, kc = yield start_new_async_kernel(kernel_name='signaltest')
+
+        def execute(cmd):
+            kc.execute(cmd)
+            reply = kc.get_shell_msg(TIMEOUT)
+            content = reply['content']
+            self.assertEqual(content['status'], 'ok')
+            return content
+
+        self.addCleanup(kc.stop_channels)
+        self.addCleanup(km.shutdown_kernel)
+        N = 5
+        for i in range(N):
+            execute("start")
+        time.sleep(1)  # make sure subprocs stay up
```

Comment:
Sleep seem to also make the test break... not sure why. Tried to replace by `yield asyncio.sleep()` to no avail.